test/box/gh-5977-inconsistency-on-shadow-build-with-hash-pk.test.lua

```diff
@@ -0,0 +1,89 @@
+env = require('test_run')
+test_run = env.new()
+fiber = require('fiber')
+
+-- Test consists of two reproducers from a corresponding issue.
+
+s = box.schema.space.create('test', {engine = 'memtx'})
+i = s:create_index('pk',  {type='hash', parts={{1, 'uint'}}})
+
+known = {}
+
+test_run:cmd("setopt delimiter ';'");
+
+function replace()
+    local r = 42
+    while known[r] do r = math.random(1000000) end
```

Comment:
The less test is randomized, the easier it will be to reproduce (in case of flaky fails for instance). Could you use pre-defined values here?

Reply:
I think `math.randomseed()` will solve this problem.

Reply:
Nevertheless still don't get why can't you simply use sequenced values..